minimal-plugin/src/main/java/io/micronaut/gradle/MicronautRuntimeDependencies.java

```diff
@@ -0,0 +1,155 @@
+ * limitations under the License.
+ */
+package io.micronaut.gradle;
+
+import org.gradle.api.plugins.JavaPlugin;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static io.micronaut.gradle.MicronautRuntime.*;
+
+/**
+ * Resolves the dependencies for the current runtime and application type
+ * @author Sergio del Amo
+ * @since 3.3.2
+ */
+public final class MicronautRuntimeDependencies {
+
+    private final static String GROUP_MICRONAUT = "io.micronaut";
+    private final static String ARTIFACT_ID_MICRONAUT_SERVER_NETTY = "micronaut-http-server-netty";
+    private final static String GROUP_MICRONAUT_SERVLET = "io.micronaut.servlet";
+    private final static String GROUP_MICRONAUT_AWS = "io.micronaut.aws";
+    private final static String GROUP_MICRONAUT_GCP = "io.micronaut.gcp";
+    private final static String GROUP_GOOGLE_CLOUD_FUNCTIONS = "com.google.cloud.functions";
+    private final static String ARTIFACT_ID_FUNCTIONS_FRAMEWORK_API = "functions-framework-api";
+
+    private final static String GROUP_MICRONAUT_ORACLE = "io.micronaut.oraclecloud";
+    private final static String ARTIFACT_ID_MICRONAUT_ORACLE_HTTP = "micronaut-oraclecloud-function-http";
+    private final static String ARTIFACT_ID_MICRONAUT_ORACLE_HTTP_TEST = "micronaut-oraclecloud-function-http-test";
+
+    private final static String ARTIFACT_ID_MICRONAUT_GCP_FUNCTION_HTTP = "micronaut-gcp-function-http";
+    private final static String ARTIFACT_ID_MICRONAUT_GCP_FUNCTION_HTTP_TEST = "micronaut-gcp-function-http-test";
+
+    private final static String GROUP_MICRONAUT_AZURE = "io.micronaut.azure";
+    private final static String ARTIFACT_ID_MICRONAUT_AZURE_FUNCTION_HTTP = "micronaut-azure-function-http";
+    private final static String ARTIFACT_ID_MICRONAUT_AZURE_FUNCTION_HTTP_TEST = "micronaut-azure-function-http-test";
+
+    private final static String ARTIFACT_ID_MICRONAUT_AWS_CUSTOM_RUNTIME = "micronaut-function-aws-custom-runtime";
+    private final static String ARTIFACT_ID_MICRONAUT_AWS_API_PROXY = "micronaut-function-aws-api-proxy";
+    private final static String ARTIFACT_ID_MICRONAUT_AWS_API_PROXY_TEST = "micronaut-function-aws-api-proxy-test";
+
+    private final static String ARTIFACT_ID_MICRONAUT_SERVLET_JETTY = "micronaut-http-server-jetty";
+    private final static String ARTIFACT_ID_MICRONAUT_SERVLET_TOMCAT = "micronaut-http-server-tomcat";
+    private final static String ARTIFACT_ID_MICRONAUT_SERVLET_UNDERTOW = "micronaut-http-server-undertow";
+    private final static String COLON = ":";
+    private static final String DEVELOPMENT_ONLY = "developmentOnly";
+
+    private MicronautRuntimeDependencies() {
+    }
+
+    /**
+     * A map of dependencies and scopes
+     * @return The dependencies and scopes
+     */
+    public static Map<String, List<String>> getDependencies(MicronautRuntime runtime, ApplicationType applicationType) {
```

Comment:
:x: I am not fan of "getter like" methods which are not getters. I'd prefer we call this `findDependenciesFor` or something similar.

Reply:
Resolved via https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/424/commits/ab1ccf102b25155c9ad6ee9c24a47d30b0ebe0c9